vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -674,23 +676,46 @@ def build_connector_meta(
                 save_to_host=False,
             )
 
-        for req_id, (req, block_ids) in self._reqs_need_save.items():
+        # NOTE: For prefill side, there might be chance that early added
+        # request is chunked prefill, so we need to check if new blocks are added
+        for req_id, new_block_id_groups, preempted in yield_req_data(scheduler_output):
+            if preempted:
+                self._request_block_ids[req_id] = []
+            if new_block_id_groups:
+                new_block_ids = (
+                    self._request_block_ids.get(req_id, []) + new_block_id_groups[0]
+                )
+                self._request_block_ids[req_id] = new_block_ids
+
+        added_reqs = []
+        for req_id, (req, _) in self._reqs_need_save.items():
             assert req.kv_transfer_params is not None
+            # don't use the block_ids from _reqs_need_save since it may be
+            # incomplete for partial prefill.
+            new_block_ids = self._request_block_ids.get(req_id, [])
+            is_partial = req.num_tokens > (len(new_block_ids) * self.block_size)
```

Comment:
can you double check this is working instead
```suggestion
            is_partial = request.num_computed_tokens < request.num_prompt_tokens
```
it's better once we switch to HMA to avoid counting blocks manually.

Reply:
Hi, @NickLucche , yeah, I was planning same thing originally, however, I can't find existing metadata in request supports me to do so:
```
 Adding request to save: cmpl-bc22c7a5-b415-4b15-9093-9b7c84fe30e9-0 with blocks: 44, is_partial=True, req.num_computed_tokens=0 req.num_prompt_tokens=942 
 Adding request to save: cmpl-bc22c7a5-b415-4b15-9093-9b7c84fe30e9-0 with blocks: 59, is_partial=False, req.num_computed_tokens=691 req.num_prompt_tokens=942 
```

I dumped one example here, for a prompt with total 942 tokens, first one is with num_computed_tokens = 0 and second with num_computed_tokens = 691.

So `request.num_computed_tokens < request.num_prompt_tokens` does not really working. I tried to find sth like `request.num_computed_tokens +  request.num_scheduled_tokens < request.num_prompt_tokens` witout luck. 

How about I added the is_partial func directly to 'class Request' ?